src/core/harpoon.ml

```diff
@@ -239,28 +239,27 @@ module Tactic = struct
          | Coverage.CovCtx _
            | Coverage.CovGoal (_, _, _) ->
             Misc.not_implemented "CovCtx impossible"
-         | Coverage.CovPatt (cG, p, tau) ->
-            let open Comp in
+         | Coverage.CovPatt (cG, pat, tau) ->
             let refine_ctx ctx = Whnf.cnormCtx (Whnf.normCtx ctx, ms) in
-            let cG = refine_ctx s.context.cG in
+            let cG = Coverage.compgctx_of_gctx cG in
             let cIH = refine_ctx s.context.cIH in
             dprintf
-              begin fun pr ->
-              pr.fmt "[harpoon-split] got pattern @[%a@]"
-                (P.fmt_ppr_cmp_pattern cD cG P.l0) p
+              begin
+                fun p ->
```

Comment:
I do however like that you renamed the outer `p` to `pat` so that the printer can remain `p`. This is consistent with the style used most often.